### PR TITLE
Support multiple multi-sig wallets on the same chain

### PR DESCRIPTION
**Pull request by Devorein**

### CODE REVIEW

File: components/bounties/components/MultiPaymentModal.tsx [hunk 1]
```diff
@@ -48,17 +51,27 @@ export default function MultiPaymentModal ({ bounties }: {bounties: BountyWithDe
     () => getSafesForAddress({ signer: signer!, chainId: chainId!, address: account! })
   );
 
-  // use first multisig wallet
-  const multiSigWallet = (safeInfos || [])[0] as SafeData | undefined;
-  const gnosisSafeAddress = multiSigWallet?.address;
-  const gnosisSafeChainId = multiSigWallet?.chainId;
+  const safeDataRecord = useMemo(() => {
```
Comment (mattcasey) [inline components/bounties/components/MultiPaymentModal.tsx:54]: this is an unecessary use of useMemo, and also it's watching the wrong variable

File: components/bounties/components/MultiPaymentModal.tsx [hunk 2]
```diff
@@ -142,10 +157,68 @@ export default function MultiPaymentModal ({ bounties }: {bounties: BountyWithDe
...
+              <Box justifyContent='space-between' gap={2} alignItems='center'>
+                <div
+                  className='octo-propertyrow'
+                  style={{
+                    height: 'fit-content',
+                    display: 'flex',
+                    alignItems: 'center'
+                  }}
+                >
+                  <div className='octo-propertyname octo-propertyname--readonly'>
```
Comment (mattcasey) [inline components/bounties/components/MultiPaymentModal.tsx:190]: Please do *not*  use focalboard classes  anywhere outside of focalboard. You shoudl use standard MUI components for field labels

File: components/bounties/components/MultiPaymentModal.tsx [hunk 3]
```diff
@@ -35,6 +36,8 @@ interface TransactionWithMetadata extends MetaTransactionData, Pick<Bounty, 'rew
 
 export default function MultiPaymentModal ({ bounties }: {bounties: BountyWithDetails[]}) {
   const [isLoading, setIsLoading] = useState(false);
+  const [gnosisSafeData, setGnosisSafeData] = useState<SafeData | undefined>(undefined);
```
Comment (mattcasey) [inline components/bounties/components/MultiPaymentModal.tsx:39]: please set the first wallet by default

File: components/bounties/components/MultiPaymentModal.tsx [hunk 4]
```diff
@@ -201,15 +274,15 @@ export default function MultiPaymentModal ({ bounties }: {bounties: BountyWithDe
...
-            <MultiPaymentButton
-              chainId={gnosisSafeChainId}
-              safeAddress={gnosisSafeAddress}
-              transactions={selectedApplicationIds.map(selectedApplicationId => applicationTransactionRecord[selectedApplicationId])}
-              onSuccess={onPaymentSuccess}
-              isLoading={isLoading}
-            />
-
-            <Typography color='secondary' variant='caption'>Safe address: {shortenHex(gnosisSafeAddress)}</Typography>
+            {(gnosisSafeChainId && gnosisSafeAddress) ? (
```
Comment (mattcasey) [inline components/bounties/components/MultiPaymentModal.tsx:277]: we should always show this button. it includes a tooltip asking the user to connect their gnosis safe if it's not connected

**Review by mattcasey — CHANGES_REQUESTED** [review]
see comments

File: components/bounties/components/MultiPaymentModal.tsx (hunk 1 above)
Comment (Devorein) [reply components/bounties/components/MultiPaymentModal.tsx:54]: Updated.

File: components/bounties/components/MultiPaymentModal.tsx (hunk 2 above)
Comment (Devorein) [reply components/bounties/components/MultiPaymentModal.tsx:190]: We are doing the same for bounty and proposal properties

File: components/bounties/components/MultiPaymentModal.tsx (hunk 3 above)
Comment (Devorein) [reply components/bounties/components/MultiPaymentModal.tsx:39]: Updated.

File: components/bounties/components/MultiPaymentModal.tsx (hunk 4 above)
Comment (Devorein) [reply components/bounties/components/MultiPaymentModal.tsx]: Updated. Previously, we used the same logic to show the multi-payment button. I've removed the extra disabled button.

**Pull request merged**